spec/core/ics-003-connection-semantics/README.md

```diff
@@ -360,13 +360,14 @@ function connOpenTry(
   proofClient: CommitmentProof,
   proofConsensus: CommitmentProof,
   proofHeight: Height,
+  proofHostConsensusState: CommitmentProof,
```

Comment:
Is there a way to signal that this is optional?

Reply:
Also my understanding is that in your implementation the proof field here is actually a marshalled header.

This is the flow from my understanding:

1. You pass in the marshalled header here in `proofHostConsensusState`
2. in your handler you check that the passed in header hashes to one of your stored header hashes
3.  you create a consensus state out of the header
4. you prove that the counterparty stored that consensus state in its client.

So this field shouldn't actually contain a `CommitmentProof` object. Instead it contains some arbitrary chain-specific data, that will be used to verify `proofConsensus` that is passed in above.

So I think instead of referring to it this way, it makes more sense to call this something like:

`hostConsensusInfo: bytes`

or something similar, and to signal in documentation that this is optional for chains to accept.

Reply:
> Is there a way to signal that this is optional?
yeah there is
> `hostConsensusInfo: bytes`

how about `hostConsensusStateProof?: bytes`

Reply:
I still think it shouldn't be named proof. Because the object in question is not actually a proof.

It is metadata that is constructing the value that will get proven agains the `consensusStateProof` which is a separate argument to this function.

Open to a different name other than `hostConsensusInfo` but I do think we should avoid `Proof` in the name since semantically it doesn't match with the way we're using `Proof` arguments in IBC spec functions

Reply:
Is there any other information a chain may request to extend the connection handshake info with? Or is this the only expected addition?

Reply:
This is the only addition